packages/app/src/cli/services/function/replay.ts

```diff
@@ -115,12 +113,75 @@ async function runFunctionRunnerWithLogInput(
   )
 }
 
+async function getRunFromIdentifier(
+  functionRunsDir: string,
+  functionHandle: string,
+  identifier: string,
+): Promise<FunctionRunData> {
+  const runPath = await findFunctionRun(functionRunsDir, functionHandle, identifier)
+  if (runPath === undefined) {
+    throw new AbortError(
+      `No log found for '${identifier}'.\nSearched ${functionRunsDir} for function ${functionHandle}.`,
+    )
+  }
+  const fileData = await readFile(runPath)
+  return JSON.parse(fileData)
+}
+
+interface LogFileMetadata {
+  namespace?: string
+  functionHandle?: string
+  identifier?: string
+}
+
+function parseLogFilename(filename: string): LogFileMetadata {
+  // Expected format: 20240522_150641_827Z_extensions_my-function_abcdef.json
+  const splitFilename = filename.split(/[_\.]/)
+
+  if (splitFilename.length < 6) {
+    return {}
```

Comment:
Would it make more sense to return `null` or `undefined` here? I don't think having all the fields be optional makes much sense because then it's possible to end up with an invalid state (e.g. `functionHandle` is present but `namespace` isn't).

Reply:
Yep, good point, that's nicer. Updated :+1: